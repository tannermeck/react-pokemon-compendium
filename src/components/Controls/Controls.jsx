import { filterDropDown } from "../../utils/utils";

function Controls({ abilities, selectedAbility, handleChange }){
    const filter = filterDropDown(abilities, selectedAbility)
    console.log('DROP', selectedAbility)
    return (
        <>
            <div>
                <h1>Filter By Ability:</h1>
                <select value={selectedAbility} onChange={(e) => handleChange(e.target.value)}>
                    <option value='all'>All</option>
                    {filter.map(category => (
                        <option key={category} value={category}>{category}</option>
                        ))}
                </select>
            </div>
            <div>
                <h1>Ability:</h1>
                <h1>{`${selectedAbility}`}</h1>
            </div>
            <div>
                <h1>Sort By Pokemon:</h1>
                <label>Ascending
                    <input type="radio" name="sort" value="ascending" />
                </label>
                <label>Descending
                    <input type="radio" name="sort" value="descending" />
                </label>
            </div>
        </>
    )
}
export default Controls;