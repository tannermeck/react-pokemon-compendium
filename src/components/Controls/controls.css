input[type="radio"]:checked {
    border: 2px solid white;
}
.sortPokemon {
    color: white;
}