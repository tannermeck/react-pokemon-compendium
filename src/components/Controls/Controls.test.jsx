import { screen, render } from '@testing-library/react';
import Controls from './Controls.jsx';

it('Should render a dropdown and radio button', async() => {
    const abilities = ['trace', 'klutz', 'damp'];
    const { container }= render(<Controls abilities={abilities} selectedAbility='damp' sortPokemon='asc' />);
    const dropDown = await screen.findByText(/Filter By Ability/i)
    expect(dropDown).toBeInTheDocument();
    expect(container).toMatchSnapshot();
})