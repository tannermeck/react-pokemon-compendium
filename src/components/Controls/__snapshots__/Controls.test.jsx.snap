// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a dropdown and radio button 1`] = `
<div>
  <section
    class="controlsContainer"
  >
    <div>
      <h1
        class="controlLabel"
      >
        Filter By Ability:
      </h1>
      <select>
        <option
          value="all"
        >
          All
        </option>
        <option
          value="trace"
        >
          trace
        </option>
        <option
          value="klutz"
        >
          klutz
        </option>
        <option
          value="damp"
        >
          damp
        </option>
      </select>
    </div>
    <div>
      <h1
        class="controlLabel"
      >
        Ability:
      </h1>
      <h1>
        damp
      </h1>
    </div>
    <div
      class="sortPokemon"
    >
      <h1
        class="controlLabel"
      >
        Sort By Pokemon:
      </h1>
      <input
        checked=""
        id="sort1"
        name="sort"
        type="radio"
        value="asc"
      />
      <label
        class="label1"
        for="sort1"
      >
        Ascending
      </label>
      <input
        id="sort2"
        name="sort"
        type="radio"
        value="desc"
      />
      <label
        for="sort2"
      >
        Descending
      </label>
    </div>
  </section>
</div>
`;
